src/Internal/Wallet/Cip30Mock.purs

```diff
@@ -210,3 +274,48 @@ mkCip30Mock pKey mSKey = do
 
 -- returns an action that removes the mock.
 foreign import injectCip30Mock :: String -> Cip30Mock -> Effect (Effect Unit)
+
+-- Helpers
+
+foreign import raisePaginateError :: forall a. Int -> Effect a
+
+foreign import _catchPaginateError
+  :: forall a. MaybeFfiHelper -> Effect a -> Effect (Maybe a)
+
+catchPaginateError :: forall a. Effect a -> Effect (Maybe a)
+catchPaginateError = _catchPaginateError maybeFfiHelper
+
+foreign import raiseAPIError :: forall a. Effect a
+
+paginateArray :: forall a. Maybe Paginate -> Array a -> Effect (Array a)
+paginateArray (Just { page, limit }) xs =
+  case page >= maxPages of
+    true -> raisePaginateError maxPages
+    false -> pure $ Array.slice startAt endAt xs
+  where
+  startAt = page * limit
+  endAt = (page + 1) * limit
+  maxPages = ceil $ (toNumber (length xs)) / (toNumber limit)
+paginateArray Nothing xs = pure xs
+
+hasEnoughAmount :: Maybe Value -> Array TransactionUnspentOutput -> Boolean
+hasEnoughAmount Nothing _ = true
+hasEnoughAmount (Just amountRequired) values = geq sumAmount amountRequired
+  where
+  utxoAmount (TransactionUnspentOutput { output }) = outputAmount output
+  outputAmount (TransactionOutput { amount }) = amount
+  sumAmount = fold $ map utxoAmount values
+
+-- Support NullOr
+
+foreign import data Null :: Type
+
+foreign import null :: Null
```

Comment:
Use null from `Literals.Null` - it has a RuntimeType instance that makes it possible to use it with `OneOf`.

Reply:
You mean this one? Looks like it implemented just the same.

https://github.com/rowtype-yoga/purescript-literals/blob/v1.0.2/src/Literals/Null.purs#L13-L13

Reply:
Except of the instance that is defined in untagged-union source

Reply:
Checked - still cannot use `oneOf`. I do not think that instance I wanted to use uses `RuntimeType` somehow.

![Uploading image.png…]()

Reply:
You forgot the `Castable a (OneOf Null a)` :

```

maybeToNullOr :: forall a. Castable a (OneOf Null a) => Maybe a -> NullOr a
maybeToNullOr (Just x) = asOneOf x
maybeToNullOr Nothing = asOneOf null
```